src/pharmpy/plugins/nlmixr/model.py

```diff
@@ -93,96 +121,166 @@ def _print_Function(self, expr):
+    add_statements(model, cg, model.statements.before_odes)
+
+    if model.statements.ode_system:
+        add_ode(model, cg)
+
+    add_statements(model, cg, model.statements.after_odes)
+
     cg.dedent()
     cg.add('})')
 
 
-def create_fit(cg, model):
-    """Create the call to fit"""
-    if [s.evaluation for s in model.estimation_steps._steps][0] is False:
-        cg.add(f'fit <- nlmixr2({model.name}, dataset, "focei")')
+def create_fit(cg: CodeGenerator, model: pharmpy.model.Model) -> None:
+    """
+    Create the call to fit for the nlmixr model with appropriate methods and datasets
+
+    Parameters
+    ----------
+    cg : CodeGenerator
+        A code object associated with the model.
+    model : pharmpy.model
+        A pharmpy.model.Model object.
+
+    Returns
+    -------
+    None
+        Modification of code object.
+
+    """
+    # FIXME : rasie error if the method does not match when evaluating
+    estimation_steps = model.estimation_steps[0]
+    if "fix_eta" in estimation_steps.tool_options:
+        fix_eta = True
+    else:
+        fix_eta = False
+
+    if [s.evaluation for s in model.estimation_steps._steps][0] is True:
+        max_eval = 0
+    else:
+        max_eval = estimation_steps.maximum_evaluations
+
+    method = estimation_steps.method
+    interaction = estimation_steps.interaction
+
+    nonmem_method_to_nlmixr = {"FOCE": "foce", "FO": "fo", "SAEM": "saem"}
+
+    if method not in nonmem_method_to_nlmixr.keys():
+        print_warning(
+            f"Estimation method {method} unknown to nlmixr2. Using 'FOCEI' as placeholder"
+        )
+        nlmixr_method = "focei"
     else:
-        cg.add(f'fit <- nlmixr2({model.name}, dataset, "posthoc")')
+        nlmixr_method = nonmem_method_to_nlmixr[method]
+
+    if interaction and nlmixr_method != "saem":
+        nlmixr_method += "i"
+
+    if max_eval is not None:
+        if max_eval == 0 and nlmixr_method not in ["fo", "foi", "foce", "focei"]:
+            nlmixr_method = "posthoc"
+            cg.add(f'fit <- nlmixr2({model.name}, dataset, est = "{nlmixr_method}"')
+        else:
+            f = f'fit <- nlmixr2({model.name}, dataset, est = "{nlmixr_method}", '
+            if fix_eta:
+                f += f'control=foceiControl(maxOuterIterations={max_eval}, maxInnerIterations=0, etaMat = etas))'
+            else:
+                f += f'control=foceiControl(maxOuterIterations={max_eval}))'
+            cg.add(f)
+    else:
+        cg.add(f'fit <- nlmixr2({model.name}, dataset, est = "{nlmixr_method}")')
+
+
+def add_evid(model: pharmpy.model.Model) -> pharmpy.model.Model:
+    temp_model = model
+    if "EVID" not in temp_model.dataset.columns:
+        temp_model.dataset["EVID"] = get_evid(temp_model)
+    return temp_model
 
 
 @dataclass
 class NLMIXRModelInternals:
     src: Optional[str] = None
     path: Optional[Path] = None
+    nonmem_control_stream: Optional = None
```

Comment:
Why is this needed? Shouldn't everything about NONMEM be forgotten after conversion?

Reply:
This is implemented to be able to convert statements within the `$DES` block. Since these are not saved in `model.statements`, they need to be saved in some other place for the nlmixr2 model. I decided to save the entire control stream from NONMEM. But as you stated, I agree that NONMEM should be more or less forgotten after conversion. 
Since the DES statements still need to be saved in one place or another, I modified the code to only save the DES statements instead of the entire control stream but I am happy to receive any other ideas or thoughts on other implementations

Reply:
Ok, I understand. Since we regenerate the code from scratch this is needed as a workaround.